File: java/services/policy-service/src/main/java/com/datagovernance/policy/entity/Policy.java
package com.datagovernance.policy.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "policies")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Policy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Dataset is required")
    @Column(nullable = false)
    private String dataset;

    @NotBlank(message = "Column name is required")
    @Column(name = "column_name", nullable = false)
    private String columnName;

    @NotBlank(message = "Rule is required")
    @Column(nullable = false)
    private String rule;

    @NotBlank(message = "Role is required")
    @Column(nullable = false)
    private String role;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: dotnet2/services/PolicyService/Models/Policy.cs
namespace PolicyService.Models
{
    public class Policy
    {
        public int Id { get; set; }
        public required string Dataset { get; set; }
        public required string Column { get; set; }
        public required string Rule { get; set; }  // "mask" or "deny"
        public required string Role { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
